### DISCUSSION TITLE
Getting Forbidden error

### DISCUSSION THREAD

**Discussion by abhi-jha** [spotify-web-api-java/spotify-web-api-java#280] [Q&A] [answered]
get top artists : https://github.com/spotify-web-api-java/spotify-web-api-java/blob/develop/examples/data/personalization/simplified/GetUsersTopArtistsExample.java

I have replaced my access token and it throws an error with Forbidden. Is there any way to set scope in the whole flow?

Spotify has scopes that can be set : https://developer.spotify.com/documentation/general/guides/authorization/scopes/#user-library-read

**Comment by dargmuesli** [comment]
Does https://github.com/spotify-web-api-java/spotify-web-api-java/discussions/276#discussioncomment-1965342 help?

**Comment by abhi-jha** [comment]
Where do I set it though?

**Comment by dargmuesli** [comment]
For the `AuthorizationCodeUriRequest` e.g. 
```Java
private static final AuthorizationCodeUriRequest authorizationCodeUriRequest = spotifyApi.authorizationCodeUri()
  .scope("user-top-read")
  .build();
```

**Comment by abhi-jha** [comment, marked as answer]
Yes, I did try it out : 

```
        final URI uri = spotifyApi.authorizationCodeUri()
          .scope("user-top-read")
          .build().execute();

        System.out.println(uri);
```

The URI returned here doesn't contain the code. It needs to be clicked on and the redirected URI has the code. Is there anyway to cacth that redirecetd URI? Do I need an http client to make the request and then capture the redirect URI?

**Reply by dargmuesli** [reply]
Please ask OAuth related questions on StackOverflow - or search for answers over there. This repository only offers support for issues related to its own code (see [CONTRIBUTING.md](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/develop/CONTRIBUTING.md#your-questions).
Also see Spotify's API documentation. They explain various authorization strategies very well. Links to that can be found in the README and multiple other issues/discussions here.
